### PR TITLE
Suggest using a pull request checklist template

### PR DESCRIPTION
**Pull request by psss**
Include a list of common things to check before completing the pull request to prevent forgetting about essential stuff.

### CODE REVIEW

**Comment by lukaszachy** [comment]
Could it be added as a PR template? https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates

**Comment by psss** [comment]
Yeah, I considered that and experimented a bit with it. However, I found out that the commit description is completely overriden by the template which is a bit unfortunate. Filed this in the following discussion:

* https://github.com/orgs/community/discussions/43392

As for now, it seems, there is no elegant solution to include both. So we have to either copy-paste the checklist or the commit description. Not completely sure which of the options is better. Let's gather some feedback before choosing.

**Comment by psss** [comment]
Vote here: I'd rather **copy-paste the commit** description.

File: docs/contribute.rst [hunk 1]
```diff
@@ -247,6 +247,26 @@ squash the changes into a single commit. Use multiple commits to
 group relevant code changes if the pull request is too large for a
 single commit.
 
+Consider pasting the following checklist (or selected items which
```
Comment (idorax) [inline docs/contribute.rst:250]: How about adding "Fixes: #<issue ID>" to the PR's description so as to associate PR with the corresponding issue?

Comment (psss) [reply docs/contribute.rst:250]: Good idea! Added in b57bad79.

**Comment by psss** [comment]
Ok, so the voting is perfectly balanced, I'll do the decision then: Let's start with including the checklist on the `Contribute` page, we can add the template later if needed/desired. Still missing at least one review to merge it...

**Pull request merged**